File: .NET/Microsoft.Recognizers.Text.Sequence/English/Parsers/GUIDParser.cs
using System;
using System.Text.RegularExpressions;
using Microsoft.Recognizers.Definitions;

namespace Microsoft.Recognizers.Text.Sequence.English
{
    public class GUIDParser : BaseSequenceParser
    {
        private static double scoreUpperLimit = 100;
        private static double scoreLowerLimit = 0;
        private static double baseScore = 100;
        private static double noBoundaryPenalty = 10;
        private static double noFormatPenalty = 10;
        private static double pureDigitPenalty = 15;
        private static string pureDigitRegex = @"^\d*$";
        private static string formatRegex = @"-";

        private static readonly Regex GuidElementRegex = new Regex(BaseGUID.GUIDRegexElement, RegexOptions.Compiled);

        public static double ScoreGUID(string textGUID)
        {
            double score = baseScore;

            Match elementMatch = GuidElementRegex.Match(textGUID);
            if (elementMatch.Success)
            {
                int startIndex = elementMatch.Groups[1].Index;
                string guidElement = elementMatch.Groups[1].Value;
                score -= startIndex == 0 ? noBoundaryPenalty : 0;
                score -= Regex.IsMatch(guidElement, formatRegex) ? 0 : noFormatPenalty;
                score -= Regex.IsMatch(textGUID, pureDigitRegex) ? pureDigitPenalty : 0;
            }

            return Math.Max(Math.Min(score, scoreUpperLimit), scoreLowerLimit) / (scoreUpperLimit - scoreLowerLimit);
        }

        public override ParseResult Parse(ExtractResult extResult)
        {
            var result = new ParseResult
            {
                Start = extResult.Start,
                Length = extResult.Length,
                Text = extResult.Text,
                Type = extResult.Type,
                ResolutionStr = extResult.Text,
                Value = ScoreGUID(extResult.Text),
            };

            return result;
        }
    }
}

File: Java/libraries/recognizers-text-sequence/src/main/java/com/microsoft/recognizers/text/sequence/english/parsers/GUIDParser.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

package com.microsoft.recognizers.text.sequence.english.parsers;

import com.microsoft.recognizers.text.ExtractResult;
import com.microsoft.recognizers.text.ParseResult;
import com.microsoft.recognizers.text.sequence.parsers.BaseSequenceParser;
import com.microsoft.recognizers.text.sequence.resources.BaseGUID;
import com.microsoft.recognizers.text.utilities.Match;
import com.microsoft.recognizers.text.utilities.RegExpUtility;

import java.util.regex.Pattern;

public class GUIDParser extends BaseSequenceParser {
    private static Double SCORE_UPPER_LIMIT = 100d;
    private static Double SCORE_LOWER_LIMIT = 0d;
    private static Double BASE_SCORE = 100d;
    private static Double NO_BOUNDARY_PENALTY = 10d;
    private static Double NO_FORMAT_PENALTY = 10d;
    private static Double PURE_DIGIT_PENALTY = 15d;
    private static String PURE_DIGIT_REGEX = "^\\d*$";
    private static String FORMAT_REGEX = "-";

    private static final Pattern GUID_ELEMENT_REGEX = Pattern.compile(BaseGUID.GUIDRegexElement);

    public static Double scoreGUID(String textGUID) {
        Double score = BASE_SCORE;

        Match[] elementMatch = RegExpUtility.getMatches(GUID_ELEMENT_REGEX, textGUID);
        if (elementMatch.length > 0) {
            Integer startIndex = elementMatch[0].index;
            String guidElement = elementMatch[0].value;
            score -= startIndex == 0 ? NO_BOUNDARY_PENALTY : 0;
            score -= Pattern.compile(FORMAT_REGEX).matcher(guidElement).find() ? 0 : NO_FORMAT_PENALTY;
            score -= Pattern.compile(PURE_DIGIT_REGEX).matcher(textGUID).find() ? PURE_DIGIT_PENALTY : 0;
        }

        return Math.max(Math.min(score, SCORE_UPPER_LIMIT), SCORE_LOWER_LIMIT)
                / (SCORE_UPPER_LIMIT - SCORE_LOWER_LIMIT);
    }

    @Override
    public ParseResult parse(ExtractResult extResult) {
        return new ParseResult(extResult.getStart(), extResult.getLength(), extResult.getText(), extResult.getType(),
                null, GUIDParser.scoreGUID(extResult.getText()), extResult.getText());
    }
}
